src/telegram.ts

```diff
@@ -21,18 +21,14 @@ class Telegram extends ApiClient {
   /**
    * Get download link to a file
    */
-  getFileLink(fileId: string | tt.File) {
-    return Promise.resolve(fileId)
-      .then((fileId) => {
-        if (typeof fileId === 'string') return this.getFile(fileId)
-        if (fileId.file_path !== undefined) return fileId
-        return this.getFile(fileId.file_id)
-      })
-      .then(
-        (file) =>
-          // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-          `${this.options.apiRoot}/file/bot${this.token}/${file.file_path}`
-      )
+  async getFileLink(fileId: string | tt.File) {
+    if (typeof fileId === 'string') {
+      fileId = await this.getFile(fileId)
+    } else if (fileId.file_path === undefined) {
```

Comment:
```suggestion
    } else if (fileId.file_path !== undefined) {
```

Reply:
In the old code,
```ts
if (fileId.file_path !== undefined) return fileId
return this.getFile(fileId.file_id)
```
made sure that `getFile` is only called when `fileId.file_path !== undefined` is `false`.

You're suggesting to call `getFile` when `fileId.file_path !== undefined` is `true`.